tests/gatewayapitests/status/gateway_test.go

```diff
@@ -1125,15 +1214,27 @@ func TestMultipleGatewaySameHostname(t *testing.T) {
 			{
 				Type:               string(gatewayv1.GatewayConditionAccepted),
 				Status:             metav1.ConditionFalse,
-				Message:            "Gateway contains 1 invalid listener(s)",
+				Message:            "Gateway does not contain any valid listener",
 				ObservedGeneration: 1,
 				Reason:             string(gatewayv1.GatewayReasonListenersNotValid),
 			},
+			{
+				Type:               string(gatewayv1.GatewayConditionProgrammed),
+				Status:             metav1.ConditionFalse,
+				Message:            "Gateway not programmed",
+				ObservedGeneration: 1,
+				Reason:             string(gatewayv1.GatewayReasonInvalid),
+			},
 		},
-		Listeners: tests.GetListenerStatusV1(ports, []int32{0, 0}),
+		Listeners: tests.GetListenerStatusV1(ports, []int32{0, 0}, false, true),
 	}
-	expectedStatus.Listeners[0].Conditions[0].Reason = string(gatewayv1.GatewayReasonListenersNotValid)
+	expectedStatus.Listeners[0].Conditions[0].Reason = string(gatewayv1.ListenerReasonInvalid)
 	expectedStatus.Listeners[0].Conditions[0].Status = metav1.ConditionFalse
+	expectedStatus.Listeners[0].Conditions[0].Message = "Hostname overlaps or is same as an existing gateway hostname"
+
+	expectedStatus.Listeners[0].Conditions[1].Reason = string(gatewayv1.ListenerReasonInvalid)
```

Comment:
Do we have to check other condition (listener being invalid) when original issue is overlapping hostnames?

Reply:
Yes. I have crosschecked with nginx implementation here.
https://github.com/nginxinc/nginx-gateway-fabric/blob/adb76bf149fb3aa98f0a85403b59c2c9d0331f21/internal/mode/static/state/conditions/conditions.go#L377 here